test/unit/tags/breadcrumbs.ts

```diff
@@ -23,55 +23,62 @@ suite('gb-breadcrumbs', Breadcrumbs, ({
 
   describe('clearRefinements()', () => {
     it('should update refinements with empty array', () => {
-      const updateRefinements = stub(tag(), 'updateRefinements');
-
+      spy = tag().update = sinon.spy((obj) => expect(obj).to.eql({ selected: [] }));
       tag().clearRefinements();
 
-      expect(updateRefinements.calledWith([])).to.be.true;
+      expect((<any>spy).called).to.be.true;
     });
   });
 
   describe('updateQueryState()', () => {
-    it('should call updateQuery()', () => {
+    it('should update originalQuery', () => {
       const originalQuery = 'red sneakers';
-      const updateQuery = stub(tag(), 'updateQuery');
-      tag().updateRefinements = () => null;
+      spy = tag().update = sinon.spy((obj) => {
+        expect(obj).to.eql({ originalQuery, selected: undefined, correctedQuery: undefined });
+      });
 
       tag().updateQueryState(<any>{ originalQuery });
 
-      expect(updateQuery.calledWith(originalQuery)).to.be.true;
+      expect((<any>spy).called).to.be.true;
     });
 
-    it('should call updateRefinements', () => {
+    it('should update refinements', () => {
       const selectedNavigation = ['a', 'b', 'c'];
-      const updateRefinements = stub(tag(), 'updateRefinements');
-      tag().updateQuery = () => null;
+      spy = tag().update = sinon.spy((obj) => {
+        expect(obj).to.eql({ originalQuery: undefined, selected: selectedNavigation, correctedQuery: undefined });
+      });
 
       tag().updateQueryState(<any>{ selectedNavigation });
 
-      expect(updateRefinements.calledWith(selectedNavigation)).to.be.true;
+      expect((<any>spy).called).to.be.true;
     });
-  });
 
-  describe('updateRefinements()', () => {
-    it('should call update with selected', () => {
-      const selected = [{ a: 'b' }];
-      const update = tag().update = spy();
+    it('should update correctedQuery', () => {
+      const correctedQuery = 'tylenol';
+      spy = tag().update = sinon.spy((obj) => {
+        expect(obj).to.eql({ originalQuery: undefined, selected: undefined, correctedQuery });
+      });
 
-      tag().updateRefinements(selected);
+      tag().updateQueryState(<any>{ correctedQuery });
 
-      expect(update.calledWith({ selected })).to.be.true;
+      expect((<any>spy).called).to.be.true;
     });
-  });
 
-  describe('updateQuery()', () => {
-    it('should call update() with originalQuery', () => {
-      const originalQuery = 'leather belt';
-      const update = tag().update = spy();
+    it('should update the whole query state', () => {
+      const queryState = {
+        originalQuery: 'tylenolt',
+        selectedNavigation: ['a', 'b', 'c'],
+        correctedQuery: 'tylenol'
+      };
+      spy = tag().update = sinon.spy((obj) => {
+        expect(obj.originalQuery).to.eql(queryState.originalQuery);
```

Comment:
simpler to state more constants and move the expectation to the end as

```
const originalQuery = 'tylenolt';
const originalQuery = 'tylenolt';
const selected = ['a', 'b', 'c'];
const correctedQuery = 'tylenol';
const queryState: any  = { originalQuery, correctedQuery, selectedNavigation: selected };
```

`expect(update.calledWith({ originalQuery, correctedQuery, selected })).to.be.true;`